packages/commons/src/events/state/index.ts

```diff
@@ -84,17 +85,53 @@ function getData(actions: Array<ActionDocument>) {
       if (!requestAction) {
         return status
       }
+      return deepMerge(status, requestAction.data)
+    }
+
+    return deepMerge(status, action.data)
+  }, {})
+}
+
+export function deepDropNulls<T extends Record<string, any>>(obj: T): T {
```

Comment:
Perhaps a comment for this function might be useful, I had a bit of trouble understanding what it does exactly